kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/json/JsonRecordWriterProvider.java

```diff
@@ -58,7 +60,9 @@ public RecordWriter getRecordWriter(final S3SinkConnectorConfig conf, final Stri
     try {
       return new RecordWriter() {
         final S3OutputStream s3out = storage.create(filename, true);
-        final JsonGenerator writer = mapper.getFactory().createGenerator(s3out);
+        final JsonGenerator writer = mapper.getFactory()
+                                         .createGenerator(s3out)
+                                         .setRootValueSeparator(new SerializedString(LINE_SEPARATOR));
```

Comment:
Are we ok if the last line in the file doesn't have a newline character? Not sure if anything would be affected by that, but based on the behavior we saw during debugging, presumably with this code the last line would be missing it.